cli/internal/turbopath/absolute_system_path.go

```diff
@@ -76,6 +76,30 @@ func (p AbsoluteSystemPath) Lstat() (os.FileInfo, error) {
 	return os.Lstat(p.ToString())
 }
 
+// Stat implements os.Stat for absolute path
+func (p AbsoluteSystemPath) Stat() (os.FileInfo, error) {
+	return os.Stat(p.ToString())
+}
+
+// Findup checks all parent directories for a file.
+func (p AbsoluteSystemPath) Findup(name RelativeSystemPath) (AbsoluteSystemPath, error) {
+	root := AbsoluteSystemPath(p.VolumeName() + string(os.PathSeparator))
+	checking := p
```

Comment:
`checking` is an `AbsoluteSystemPath` which means that it will never be `~/foo/` ... it can only ever be `/Users/nathanhammond/foo` or `C:\Windows`. That's a constraint which, though not enforced, is assumed by our types. (You can have garbage-in, but as long as it _isn't_ garbage-in the code here will work.)

Reply:
Sure, but this still applies. If we have `/Users/nathanhammond/foo/..` and `/Users/nathanhammond` then these are the same paths canonically, but they won't be equal under the equality check in the for loop.

Reply:
I _just_ figured out that `resolve` was supposed to be a method. 😅 Go's method name is `Clean` and it gets called for us on the first line of the loop as part of `Join`, and is set back at the tail end of the loop.